src/m365/spo/commands/site/site-recyclebinitem-restore.ts

```diff
@@ -31,31 +31,31 @@ class SpoSiteRecycleBinItemRestoreCommand extends SpoCommand {
 
     const requestUrl: string = `${args.options.siteUrl}/_api/site/RecycleBin/RestoreByIds`;
 
-    const ids: string[] = this.splitIdsList(args.options.ids);
+    const ids: string[] = Utils.splitAndTrim(args.options.ids);
     const idsChunks: string[][] = [];
 
     while (ids.length) {
       idsChunks.push(ids.splice(0, 20));
     }
 
-    idsChunks.forEach(async (idsChunk: string[], index: number) => {
-      const requestOptions: any = {
-        url: requestUrl,
-        headers: {
-          'accept': 'application/json;odata=nometadata'
-        },
-        responseType: 'json',
-        data: {
-          ids: idsChunk
-        }
-      };
-
-      await request.post(requestOptions);
-
-      if(index === idsChunks.length - 1) {
-        cb();
-      }
-    });
+    Promise.all(
+      idsChunks.map(async (idsChunk: string[]) => {
+        const requestOptions: any = {
+          url: requestUrl,
+          headers: {
+            'accept': 'application/json;odata=nometadata',
+            'content-type': 'application/json'
+          },
+          responseType: 'json',
+          data: {
+            ids: idsChunk
+          }
+        };
+
+        await request.post(requestOptions);
+      })
+    ).then(_ => cb());
```

Comment:
We should add error handling here, in case one of the promises fails.

Reply:
ok added something with some general message when debug mode

Reply:
We have a standard way of handling rejected promises, for example: https://github.com/pnp/cli-microsoft365/blob/77249d9c445638bd6e961e9698bd0b95985f72d1/src/m365/aad/commands/app/app-add.ts#L120. Let's use is here as well for consistency